proto/rollkit/rollkit.proto

```diff
@@ -49,19 +49,19 @@ message Header {
 	// pubkey can't be recovered by the signature (e.g. ed25519).
 	bytes proposer_address = 10;
 
+	//public key of the proposer so we can verify the signature
+	bytes proposer_public_key = 11;
+
 
 	// Hash of block aggregator set, at a time of block creation
-	bytes aggregators_hash = 11;
+	bytes aggregators_hash = 12;
 
 	// Chain ID the block belongs to
-	string chain_id = 12;
+	string chain_id = 13;
 }
 
 message Commit {
-	uint64 height = 1;
-	bytes header_hash = 2;
-	// Note: most of the time this will be a single sinature
-	repeated bytes signatures = 3;
+	repeated bytes signatures = 1;
 }
 
 message SignedHeader {
```

Comment:
i was wondering if we don't modify the `Header` with additional fields like public key, but just add signature and public key directly to `SignedHeader`, just like the way celestia-node does it for their `ExtendedHeader`?
```
message ExtendedHeader {
  tendermint.types.Header header = 1;
  tendermint.types.Commit commit = 2;
  tendermint.types.ValidatorSet validator_set = 3;
}
```
similarly, we could do
```
message SignedHeader {
   Header header = 1;
   repeated bytes signatures = 2;
   repeated bytes signer_public_keys = 3;
}
```
or we could retain the tendermint types `Commit` and `ValidatorSet`
```
message SignedHeader {
   Header header = 1;
   tendermint.types.Commit commit = 2;
   tendermint.types.ValidatorSet validator_set = 3;
}
```
retaining tendermint types would help with less internal conversions.

Reply:
I like the proposal. 2 Things i would like to add: 

1. We should try not to add tendermint types if we can keep them out. 

2. The `repeated bytes signer_public_keys = 3` must be inside the header as we must sign over the validator set changes.

Reply:
> 2\. The `repeated bytes signer_public_keys = 3` must be inside the header as we must sign over the validator set changes.

there is already `AggregatorsHash` part of the header which being signed and it is computed as hash of the validatorset.

Reply:
I propose we remove the hash and add the validator pub keys to the header. We are gossiping about everything anyways, so I don't see an overhead here. There are definitely improvements /optimizations possible that we could follow up on later.